dbuf-gen/src/kotlin_gen/generate.rs

```diff
@@ -0,0 +1,364 @@
+use pretty::{BoxAllocator, BoxDoc, DocAllocator};
+
+use crate::ast::{Module, Type, TypeKind};
+
+mod kotlin {
```

Comment:
I don't think this kind of nesting is needed here, everything in this module is already under `kotlin_gen::generate`